app/src/main/java/com/android/shelfLife/model/newRecipe/RecipeRepositoryFirestore.kt

```diff
@@ -31,144 +33,202 @@ class RecipeRepositoryFirestore @Inject constructor(private val db: FirebaseFire
   private val _selectedRecipe = MutableStateFlow<Recipe?>(null)
   override val selectedRecipe: StateFlow<Recipe?> = _selectedRecipe.asStateFlow()
 
+  private var recipesListenerRegistration: ListenerRegistration? = null
```

Comment:
please do the listener the way it's implemented inside the invitations, this implementation doesn't work

Reply:
This can be found on my branch as the invitations on main don't work completely yet